okhttp/src/jvmTest/java/okhttp3/internal/idn/IdnaMappingTableTest.kt

```diff
@@ -46,11 +58,49 @@ class IdnaMappingTableTest {
     assertThat("™".map()).isEqualTo("tm")
   }
 
+  /** Confirm the compact table satisfies is documented invariants. */
+  @Test fun validateCompactTableInvariants() {
```

Comment:
TODO: confirm the output sizes are smaller than the ranges we use to index into them.

sections - at most 16,384 bytes (14 bits, which is what we binary search on.)
ranges - at most 65,536 bytes. (14 bits * 4, which is what we index on with a precision of 4 bytes.)
mappings - at most 16,384 chars (14 bits, which is what we index on.)

Reply:
done.